master/static/srv/task-logging-setup.sh

```diff
@@ -51,41 +51,40 @@ if [ -n "$DET_K8S_LOG_TO_FILE" ]; then
     ((DET_LOG_WAIT_COUNT += 2))
 fi
 
-if [ "$DET_RESOURCES_TYPE" == "slurm-job" ] || [ "$DET_NO_FLUENT" == "true" ]; then
-    export PATH="/run/determined/pythonuserbase/bin:$PATH"
-    if [ -z "$DET_PYTHON_EXECUTABLE" ]; then
-        export DET_PYTHON_EXECUTABLE="python3"
-    fi
+export PATH="/run/determined/pythonuserbase/bin:$PATH"
+if [ -z "$DET_PYTHON_EXECUTABLE" ]; then
+    export DET_PYTHON_EXECUTABLE="python3"
+fi
 
-    if ! /bin/which "$DET_PYTHON_EXECUTABLE" >/dev/null 2>&1; then
-        echo "{\"log\": \"error: unable to find python3 as '$DET_PYTHON_EXECUTABLE'\n\", \"timestamp\": \"$(date --rfc-3339=seconds)\"}" >&2
-        echo "{\"log\": \"please install python3 or set the environment variable DET_PYTHON_EXECUTABLE=/path/to/python3\n\", \"timestamp\": "$(date --rfc-3339=seconds)"}" >&2
-        exit 1
-    fi
+if ! "$DET_PYTHON_EXECUTABLE" --version >/dev/null 2>&1; then
+    echo "{\"log\": \"error: unable to find python3 as '$DET_PYTHON_EXECUTABLE'\n\", \"timestamp\": \"$(date --rfc-3339=seconds)\"}" >&2
+    echo "{\"log\": \"please install python3 or set the environment variable DET_PYTHON_EXECUTABLE=/path/to/python3\n\", \"timestamp\": "$(date --rfc-3339=seconds)"}" >&2
+    export PYTHON_MISSING_EXIT_CODE=8
+    exit $PYTHON_MISSING_EXIT_CODE
```

Comment:
i think for this PR, just exit 1, and it will be in the logs. dont want to have to wire reading this up for all RMs today.

Reply:
The reason I wired it up was because my task containers were getting removed before I had a chance to read the logs in the task container so all I was seeing was an "exited with a nonzero exit code: 1"; however, I can see why this wouldn't be an issue if those logs were persisted after the container was removed